src/controller/tests/TestWriteChunking.cpp

```diff
@@ -454,13 +478,149 @@ void TestWriteChunking::TestNonConflictWrite(nlTestSuite * apSuite, void * apCon
+        status.push_back(PathStatus(aPath, aError));
+        onGoingPath = ConcreteAttributePath();
+        ChipLogProgress(Zcl, "OnListWriteEnd endpoint=%" PRIu16 " Cluster=" ChipLogFormatMEI " attribute=" ChipLogFormatMEI,
+                        aPath.mEndpointId, ChipLogValueMEI(aPath.mClusterId), ChipLogValueMEI(aPath.mAttributeId));
+    };
+
+    ByteSpan list[kTestListLength];
+
+    for (const auto & p : instructions.paths)
+    {
+        err = writeClient->EncodeAttribute(AttributePathParams(p.mEndpointId, p.mClusterId, p.mAttributeId),
+                                           DataModel::List<ByteSpan>(list, kTestListLength));
+        NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+    }
+
+    err = writeClient->SendWriteRequest(sessionHandle);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    ctx.GetIOContext().DriveIOUntil(System::Clock::Seconds16(10),
+                                    [&]() { return ctx.GetExchangeManager().GetNumActiveExchanges() == 0; });
+
+    NL_TEST_ASSERT(apSuite, onGoingPath == app::ConcreteAttributePath());
+    NL_TEST_ASSERT(apSuite, status.size() == instructions.expectedStatus.size());
+
+    for (size_t i = 0; i < status.size(); i++)
+    {
+        NL_TEST_ASSERT(apSuite, status[i] == PathStatus(instructions.paths[i], instructions.expectedStatus[i]));
+    }
+
+    testServer.mOnListWriteBegin = nullptr;
+    testServer.mOnListWriteEnd   = nullptr;
+}
+
+} // namespace TestTransactionalListInstructions
+
+void TestWriteChunking::TestTransactionalList(nlTestSuite * apSuite, void * apContext)
+{
+    using namespace TestTransactionalListInstructions;
+
+    TestContext & ctx = *static_cast<TestContext *>(apContext);
+
+    // Initialize the ember side server logic
+    InitDataModelHandler(&ctx.GetExchangeManager());
+
+    // Register our fake dynamic endpoint.
+    emberAfSetDynamicEndpoint(0, kTestEndpointId, &testEndpoint, 0, 0, Span<DataVersion>(dataVersionStorage));
+
+    // Register our fake attribute access interface.
+    registerAttributeAccessOverride(&testServer);
+
+    // Test 1: we should receive transaction notifications
+    ChipLogProgress(Zcl, "Test 1: we should receive transaction notifications");
+    RunTest(apSuite, ctx,
+            Instructions{
+                .paths          = { ConcreteAttributePath(kTestEndpointId, Clusters::TestCluster::Id, kTestListAttribute) },
+                .expectedStatus = { CHIP_NO_ERROR },
+            });
+
+    ChipLogProgress(Zcl, "Test 2: we should receive transaction notifications for incomplete list operations");
+    RunTest(apSuite, ctx,
+            Instructions{
+                .paths            = { ConcreteAttributePath(kTestEndpointId, Clusters::TestCluster::Id, kTestListAttribute) },
+                .onListWriteBegin = [&](std::unique_ptr<WriteClient> & writeClient,
+                                        const app::ConcreteAttributePath & aPath) { writeClient = nullptr; },
```

Comment:
Explain what setting `writeClient to null` does?

Reply:
Used an enum for a more clear return value.